docs/index.md

```diff
@@ -55,3 +55,28 @@ sequenceDiagram
     end
     App->>User: send response
 ```
+
+### Receipts
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant User
+    participant Proxy
+    participant App
+
+    note over User: Make mutating request so <br>have request and response
+
+    User->>Proxy: /etcdserverpb.Receipt/GetReceipt
+    Proxy->>App: /etcdserverpb.Receipt/GetReceipt
+    App->>Proxy: 202 Accepted, retry-after: 3s
+    Proxy->>User: 202 Accepted, retry-after: 3s
+    note over Proxy: A smart proxy may instead <br>handle the retry internally
```

Comment:
I think that in this case, the proxy has to be a client library since client<->server communications encrypted over TLS?

Reply:
Perhaps `Proxy` is the wrong word for this. I'm imagining something similar to the Kubernetes API Server where clients have to make a request to the server and then the API Server instances make a request to the datastore to persist things (and query missing state). In this case the client might not trust the API Server to actually be doing things so receipts could help verify that transactions are actually executed.